libs/coin-framework/src/serialization/account.ts

```diff
@@ -289,6 +292,11 @@ function toTokenAccountRaw(
     approvals,
   } = ta;
 
+  let starred;
+  if (userData) {
```

Comment:
mmh, the concerns are already separated actually.

if you look at LLC, we have this:

```
export function toAccountRaw(account: Account, userData?: AccountUserData): AccountRaw {
  const bridge = getAccountBridge(account);
  return commonToAccountRaw(
    account,
    {
      assignToAccountRaw: bridge.assignToAccountRaw,
      toOperationExtraRaw: bridge.toOperationExtraRaw,
    },
    userData,
  );
}
```

Here, I have applied the exact same idea as what we do for connecting extra data (here we have similar topic for the coin specific of assignToAccountRaw and toOperationExtraRaw)

and the same applied to "userData" which is basically just an object of type: 
```
type AccountUserData = {
  // the Account#id
  id: string;
  // user's name for this account
  name: string;
  // user's starred account ids: it can be more than the account.id because token accounts can also be starred
  starredIds: string[];
};
```

so i'm not sure exactly what you want me to rework this into, but coin-framework **do not depends on live-wallet**, only LLC does, and the glue is done in LLC exactly like for coins.

Reply:
also, this answers a bit for the concerns about the Account <> AccountRaw topic : today Account is already not 1:1 with AccountRaw since we have coin specifics fields in Account that are managed on top of the raw implementation and come as extra data (here as a second parameter)
i just applied the same pattern on a third parameter.

Reply:
Ok for this last point ; nevertheless it could be great to think of a dedicated function to manage current and future update of the saved model.

For the first point, my concern is about the logic in CoinFramework that "knows" what a "userData" is, and therefore apply a modification on the "starred" property. In fact, this one, as the userData, are truly unknown by the CoinFramework. Those properties are pure UI concerns.

Reply:
i agree that user data is UI concerns but also ultimately so is "AccountRaw".  coin-framework shouldn't know about AccountRaw, it's a data format used by the UI to store all the data together (and it's a bit implementation detail specific)